src/main.js

```diff
@@ -332,8 +332,16 @@ function grammarsFromScriptElements(optNodeOrNodeList) {
   return ns;
 }
 
-function makeRecipe(recipeFn) {
-  return recipeFn.call(new Builder());
+function makeRecipe(recipe) {
+  if (typeof recipe === 'function') {
+    return recipe.call(new Builder());
+  } else {
```

Comment:
If it is an object, it would count as a deserialized JSON recipe. But maybe it's cleaner to only allow "stringified" JSON recipes with `makeRecipe`.